<!-- -*- mode: XML -*- -->
<!-- run laser_object_detector/LaserObjectDetectorNodelet in a nodelet manager -->

<launch>
    <arg name="start_as_nodelet" default="true" />

    <arg name="manager" default="velodyne_nodelet_manager" />
    <arg name="input_topic" default="/velodyne_points" />
    <arg name="input_is_velodyne" default="true" />
    <arg name="circular_buffer_capacity" default="6000" />
    <arg name="angle_between_scanpoints" default="0.2" />

    <arg name="certainty_threshold" default="0.0" />
    <arg name="dist_weight" default="0.75" />
    <arg name="intensity_weight" default="0.25" />
    <arg name="object_size_in_m" default="1.2" />
    <arg name="distance_to_comparison_points" default="0.15" />

    <arg name="median_min_dist" default="0.15" />
    <arg name="median_thresh1_dist" default="0.45" />
    <arg name="median_thresh2_dist" default="7.0" />
    <arg name="median_max_dist" default="10.0" />

    <arg name="kernel_size_diff_factor" default="9.0" />
    <arg name="max_dist_for_median_computation" default="0.0" />
    <arg name="max_kernel_size" default="100" />
    <arg name="publish_debug_cloud" default="false" />
    <arg name="output_topic" default="/laser_detector_objects" />

    <node unless="$(arg start_as_nodelet)" launch-prefix="" pkg="laser_object_detector" type="laser_object_detector_node" name="laser_object_detector"
          output="screen" >
        <param name="input_topic" value="$(arg input_topic)"/>
        <param name="input_is_velodyne" value="$(arg input_is_velodyne)"/>
        <param name="angle_between_scanpoints" value="$(arg angle_between_scanpoints)"/>
        <param name="circular_buffer_capacity" value="$(arg circular_buffer_capacity)"/>

        <param name="certainty_threshold" value="$(arg certainty_threshold)"/>
        <param name="dist_weight" value="$(arg dist_weight)"/>
        <param name="intensity_weight" value="$(arg intensity_weight)"/>
        <param name="object_size_in_m" value="$(arg object_size_in_m)"/>
        <param name="distance_to_comparison_points" value="$(arg distance_to_comparison_points)"/>

        <param name="median_min_dist" value="$(arg median_min_dist)"/>
        <param name="median_thresh1_dist" value="$(arg median_thresh1_dist)"/>
        <param name="median_thresh2_dist" value="$(arg median_thresh2_dist)"/>
        <param name="median_max_dist" value="$(arg median_max_dist)"/>

        <param name="kernel_size_diff_factor" value="$(arg kernel_size_diff_factor)"/>
        <param name="max_dist_for_median_computation" value="$(arg max_dist_for_median_computation)"/>
        <param name="max_kernel_size" value="$(arg max_kernel_size)"/>
        <param name="publish_debug_cloud" value="$(arg publish_debug_cloud)"/>

	    <remap from="/laser_detector_objects" to="$(arg output_topic)" />
    </node>

    <node if="$(arg start_as_nodelet)" pkg="nodelet" type="nodelet" name="laser_object_detector_nodelet"
          args="load laser_object_detector/laser_object_detector_nodelet $(arg manager)" output="screen" >
        <param name="input_topic" value="$(arg input_topic)"/>
        <param name="input_is_velodyne" value="$(arg input_is_velodyne)"/>
        <param name="angle_between_scanpoints" value="$(arg angle_between_scanpoints)"/>
        <param name="circular_buffer_capacity" value="$(arg circular_buffer_capacity)"/>

        <param name="certainty_threshold" value="$(arg certainty_threshold)"/>
        <param name="dist_weight" value="$(arg dist_weight)"/>
        <param name="intensity_weight" value="$(arg intensity_weight)"/>
        <param name="object_size_in_m" value="$(arg object_size_in_m)"/>
        <param name="distance_to_comparison_points" value="$(arg distance_to_comparison_points)"/>

        <param name="median_min_dist" value="$(arg median_min_dist)"/>
        <param name="median_thresh1_dist" value="$(arg median_thresh1_dist)"/>
        <param name="median_thresh2_dist" value="$(arg median_thresh2_dist)"/>
        <param name="median_max_dist" value="$(arg median_max_dist)"/>

        <param name="kernel_size_diff_factor" value="$(arg kernel_size_diff_factor)"/>
        <param name="max_dist_for_median_computation" value="$(arg max_dist_for_median_computation)"/>
        <param name="max_kernel_size" value="$(arg max_kernel_size)"/>
        <param name="publish_debug_cloud" value="$(arg publish_debug_cloud)"/>

	    <remap from="/laser_detector_objects" to="$(arg output_topic)" />
    </node>

</launch>
